### PR TITLE
[FLINK-16874] Respect the dynamic options when calculating memory options in taskmanager.sh

### PR DESCRIPTION
**Pull request by KarmaGYZ**
## What is the purpose of the change

Let taskmanager.sh respect the dynamic options when calculating memory options.

## Brief change log

Pass dynamic args when calculating memory options through BashJavaUtils.

## Verifying this change

One could run something like "./bin/taskmanager.sh start -Dtaskmanager.memory.process.size=4096m" to verify whether the dynamic options take effect.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by KarmaGYZ** [comment]
Travis passed. https://travis-ci.com/github/flink-ci/flink/builds/157011124

**Comment by KarmaGYZ** [comment]
cc @azagrebin

File: flink-dist/src/main/flink-bin/bin/config.sh [hunk 1]
```diff
@@ -618,10 +618,11 @@ TMSlaves() {
 runBashJavaUtilsCmd() {
     local cmd=$1
     local conf_dir=$2
-    local class_path="${3:-$FLINK_BIN_DIR/bash-java-utils.jar:`findFlinkDistJar`}"
+    local class_path=$3
+    local dynamic_args=${@:4}
     class_path=`manglePathList ${class_path}`
 
-    local output=`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} 2>&1 | tail -n 1000`
+    local output=`${JAVA_RUN} -classpath ${class_path} org.apache.flink.runtime.util.BashJavaUtils ${cmd} --configDir ${conf_dir} $dynamic_args 2>&1 | tail -n 1000`
```
Comment (azagrebin) [inline flink-dist/src/main/flink-bin/bin/config.sh:625]: To make sure that it works, I suggest to add a test to `BashJavaUtilsITCase` where we rewrite some memory options in `flink-conf.yaml` by dynamic args and check the result.

**Comment by KarmaGYZ** [comment]
Thanks for the review @azagrebin . I've updated the PR.

File: flink-dist/src/test/java/org/apache/flink/dist/BashJavaUtilsITCase.java [hunk 2]
```diff
@@ -46,4 +50,15 @@ public void testGetTmResourceParamsConfigs() throws Exception {
...
 	}
+
+	@Test
+	public void testConfigOverwrittenByDynamicOpts() throws Exception {
+		double cpuCores = 39.0;
+		String[] commands = {RUN_BASH_JAVA_UTILS_CMD_SCRIPT, BashJavaUtils.Command.GET_TM_RESOURCE_PARAMS.toString(), "-D" + TaskManagerOptions.CPU_CORES.key() + "=" + cpuCores};
+		List<String> lines = Arrays.asList(executeScript(commands).split(System.lineSeparator()));
+
+		assertEquals(2, lines.size());
+		Map<String, String> configs = ConfigurationUtils.parseTmResourceDynamicConfigs(lines.get(lines.size() - 1));
```
Comment (azagrebin) [inline flink-dist/src/test/java/org/apache/flink/dist/BashJavaUtilsITCase.java:61]:
```suggestion
		assertThat(lines.size(), is(2));
		Map<String, String> configs = ConfigurationUtils.parseTmResourceDynamicConfigs(lines.get(1));
```

Comment (KarmaGYZ) [reply flink-dist/src/test/java/org/apache/flink/dist/BashJavaUtilsITCase.java]: Make sense. Fixed.

**Pull request closed without merging**